README.md

```diff
@@ -1,2 +1,232 @@
-# go-libp2p-rcmgr
-An implmentation of the libp2p network resource manager interface
+# The libp2p Network Resource Manager
+
+This package contains the canonical implementation of the libp2p
+Network Resource Manager interface.
+
+The implementation is based on the concept of Resource Management
+Scopes, whereby resource usage is constrained by a DAG of scopes,
+accounting for multiple levels of resource constraints.
+
+## Design Considerations
+
+- The Resource Manager must account for basic resource usage at all
+  levels of the stack, from the internals to the application.
```

Comment:
I'm not disagreeing with this, but it may be useful to define what is meant by application and provide some more justification why libp2p resource manager interface bring the application concerns in scope.

Reply:
yeah, it is meant to cover network (libp2p) usage by the application; will add some text to clarify.